bodhi/server/models.py

```diff
@@ -2846,7 +2848,8 @@ def __str__(self):
         val += u"""
   Submitter: %s
   Submitted: %s\n""" % (username, self.date_submitted)
-        if len(self.comments):
+        # Counting length of a generator object {self.comments_since_karma_reset}
+        if sum(1 for _ in self.comments_since_karma_reset):
```

Comment:
PEP8 recommends the following 
```
Yes: if not seq:
        if seq:

No: if len(seq):
       if not len(seq):
```
So we should drop the `if len()`/`if sum()` here. 

I would recommend converting `self.comments_since_karma_reset` into a list and store it in a variable (for example `comments_since_karma_reset`) then use this variable in the if statement.